### PR TITLE
Adds fallback support for docker-compose

### PR DESCRIPTION
**Pull request by dcode**
Open to discuss the changes and/or discard if it's not desired. 

- Still prefers `docker compose` command via the docker plugin (bundled with Docker Desktop)
- fallback to `docker-compose` allows for using podman or docker ce
- also adds selinux flags for shared volumes, should be ignored where not supported
- Formats using shell-format and cleans up shellcheck problems

### CODE REVIEW

**Comment by peasead** [comment]
On macOS, this outputs this but continues. I assume the check for `docker-compose` vs. `docker-compose-plugin` isn't "failing" silently?

Also a `WARN` on `WARN[0000] mount of type `volume` should not define `bind` option`.
```
sh elastic-container.sh start

Usage:  docker compose [OPTIONS] COMMAND

Docker Compose

Options:
      --ansi string                Control when to print ANSI control characters ("never"|"always"|"auto") (default "auto")
      --compatibility              Run compose in backward compatibility mode
      --env-file string            Specify an alternate environment file.
  -f, --file stringArray           Compose configuration files
      --profile stringArray        Specify a profile to enable
      --project-directory string   Specify an alternate working directory
                                   (default: the path of the, first specified, Compose file)
  -p, --project-name string        Project name

Commands:
  build       Build or rebuild services
  convert     Converts the compose file to platform's canonical format
  cp          Copy files/folders between a service container and the local filesystem
  create      Creates containers for a service.
  down        Stop and remove containers, networks
  events      Receive real time events from containers.
  exec        Execute a command in a running container.
  images      List images used by the created containers
  kill        Force stop service containers.
  logs        View output from containers
  ls          List running compose projects
  pause       Pause services
  port        Print the public port for a port binding.
  ps          List containers
  pull        Pull service images
  push        Push service images
  restart     Restart service containers
  rm          Removes stopped service containers
  run         Run a one-off command on a service.
  start       Start services
  stop        Stop services
  top         Display the running processes
  unpause     Unpause services
  up          Create and start containers
  version     Show the Docker Compose version information

Run 'docker compose COMMAND --help' for more information on a command.
Passphrase has been reset. Proceeding.
Starting Elastic Stack network and containers
WARN[0000] mount of type `volume` should not define `bind` option
[+] Running 4/5
[+] Running 6/5tic-container_default       Created                                                                                                                              0.0s
[+] Running 7/8tic-container_default       Created                                                                                                                              0.0s
 ⠿ Network elastic-container_default       Created
...
```

File: elastic-container.sh [hunk 1]
```diff
@@ -3,20 +3,38 @@ set -o pipefail
...
+declare MacOSDR
+
+# Ignore following warning
+# shellcheck disable=SC1091
 . .env
 
+declare COMPOSE
+if docker compose 2>/dev/null; then
+  COMPOSE="docker compose"
+elif command -v docker-compose >/dev/null; then
```
Comment (DefSecSentinel) [inline elastic-container.sh:18]:
```suggestion
if docker compose >/dev/null 2>&1; then
  COMPOSE="docker compose"
elif command -v docker-compose >/dev/null 2>&1; then
```

Comment (dcode) [reply elastic-container.sh:18]: I don't want to throw this completely away, so I'll move it until after the verbose check

**Comment by DefSecSentinel** [comment]
Any idea what this is from? 

WARN[0000] mount of type `volume` should not define `bind` option

**Comment by DefSecSentinel** [comment]
Is it from the :z you appended to the volume declarations?

**Comment by dcode** [comment]
yes, it's from the :z flag, but that's required on Linux, even if using
Docker Desktop. Did the stderr redirect not fix it?

**Pull request merged**